csharp/src/Google.Protobuf/Reflection/MethodDescriptor.cs

```diff
@@ -125,23 +121,23 @@ public RepeatedField<T> GetOption<T>(RepeatedExtension<MethodOptions, T> extensi
         /// <summary>
         /// The brief name of the descriptor's target.
         /// </summary>
-        public override string Name { get { return proto.Name; } }
+        public override string Name => Proto.Name;
 
         internal void CrossLink()
         {
             IDescriptor lookup = File.DescriptorPool.LookupSymbol(Proto.InputType, this);
-            if (!(lookup is MessageDescriptor))
+            if (lookup is not MessageDescriptor md)
```

Comment:
Maybe rename `md` and `md2` to `inputType` and `outputType`? That way the assignments look more natural IMO.